crategen/models.py

```diff
@@ -0,0 +1,279 @@
+        name (str): The name of the TES task.
+        description (Optional[str]): A brief description of the TES task.
+        executors (list[Executor]): The executors associated with the TES task.
+        inputs (list[TESInputs]): The inputs to the TES task.
+        outputs (list[TESOutputs]): The outputs of the TES task.
+        creation_time (str): The time the task was created.
+        logs (list[TESLogs]): Logs associated with the TES task.
+    """
+    id: str
+    name: str
+    description: Optional[str] = ""
+    executors: list[Executor]
+    inputs: list[TESInputs]
+    outputs: list[TESOutputs]
+    creation_time: str
+    logs: list[TESLogs]
+
+    class Config:
+        extra = "forbid"
+
+
+class WESRunLog(BaseModel):
+    """
+    A model representing a run log in the Workflow Execution Service (WES).
+
+    Attributes:
+        name (Optional[str]): The name of the run.
+        start_time (Optional[str]): The start time of the run.
+        end_time (Optional[str]): The end time of the run.
+        cmd (Optional[list[str]]): The command executed in the run.
+        stdout (Optional[str]): The path to the stdout log.
+        stderr (Optional[str]): The path to the stderr log.
+        exit_code (Optional[int]): The exit code of the run.
+    """
+    name: Optional[str] = None
+    start_time: Optional[str] = None
+    end_time: Optional[str] = None
+    cmd: Optional[list[str]] = None
+    stdout: Optional[str] = None
+    stderr: Optional[str] = None
+    exit_code: Optional[int] = None
+
+
+class WESOutputs(BaseModel):
+    """
+    A model representing output files in WES.
+
+    Attributes:
+        location (str): The URL of the output file.
+        name (str): The name of the output file.
+    """
+    location: str
+    name: str
+
+
+class WESRequest(BaseModel):
+    """
+    A model representing a workflow request in WES.
+
+    Attributes:
+        workflow_params (dict[str, str]): The parameters for the workflow.
+        workflow_type (str): The type of the workflow (e.g., CWL).
+        workflow_type_version (str): The version of the workflow type.
+        tags (Optional[dict[str, str]]): Additional tags associated with the workflow.
+    """
+    workflow_params: dict[str, str]
+    workflow_type: str
+    workflow_type_version: str
+    tags: Optional[dict[str, str]] = None
+
+
+class WESData(BaseModel):
+    """
+    A model representing a WES run.
+
+    Attributes:
+        run_id (str): The unique identifier for the WES run.
+        request (WESRequest): The request associated with the WES run.
+        state (str): The state of the WES run.
+        run_log (WESRunLog): The log of the WES run.
+        task_logs (Optional[list[WESRunLog]]): The logs of individual tasks within the run.
+        outputs (list[WESOutputs]): The outputs of the WES run.
+    """
+    run_id: str
+    request: WESRequest
+    state: str
+    run_log: WESRunLog
+    task_logs: Optional[list[WESRunLog]] = None
```

Comment:
According to the Ga4GH WES specification this field has been deprecated and replaced with the `tes_logs_url` field. You can still include it to support current and older versions, but perhaps we can consider adding a deprecation warning as well as adding support for `tes_logs_url` feild.

Reference: https://ga4gh.github.io/workflow-execution-service-schemas/docs/#tag/Workflow-Runs/operation/GetRunLog

Reply:
I'll include support for the `tes_logs_url` field in the `WESData` model and add a deprecation warning for the `task_logs` field, following the GA4GH WES specification.